src/screens/home/Home.tsx

```diff
@@ -69,6 +85,44 @@ const Home = ({ navigation, route }: NativeStackScreenProps<HomeStackParams>) =>
         }
     };
 
+    const attemptFetchUserDoc = async (user: any, attempts = 0) => {
+        const MAX_ATTEMPTS = 15
+        try {
+            const userDocRef = doc(db, "users", user.uid);
+            const userDoc = await getDoc(userDocRef);
+
+            if (!userDoc.exists()) {
+                if (attempts < MAX_ATTEMPTS) {
+                    console.warn(`Attempt ${attempts + 1} failed. Retrying...`);
+                    await attemptFetchUserDoc(user, attempts + 1);
```

Comment:
Why is this recursive?

Reply:
Ohh the logic for this was really bad.
but fixed now
[4411e2f](https://github.com/TAMUSHPE/MobileApp/pull/490/commits/4411e2f84c8dea4ee3a0ecb14aa1e5e28d2df34f)

refactor with extracted fetchAndStoreUserFunction
fb17f01e8c2445b012eedb50679bd81547704b3d